Reader: update following link in discover stream

jhnstn opened the pull request:
Part of #7057

~~This updates the follow url in the discover stream to use the source blog instead of discover.wordpress.com.~~

~~It hides the follow option if the pick is not from wpcom:~~
~~**Update** it disables the follow option if the site is not on wpcom:~~

**Update** This adds inline follow buttons to the discover posts.
The follow button will on only display on wpcom sites.


/cc @jancavan , @bluefuton 

Test live: https://calypso.live/?branch=update/reader/discover-stream-following

blowery commented:
I find this really confusing. The follow button on each card is supposed to be for the site you're viewing, and in this case, that's Discover.

Could we put a Follow button next to the site name in the attribution on the card instead? That would at least visually tie it to the site.

jhnstn commented:
I guess I was thinking of the discover stream as behaving more like recommended posts where each post is linked to a different site. 

But it's no problem to add the inline follow button

jancavan commented:
Mentioned in https://github.com/Automattic/wp-calypso/issues/7057#issuecomment-235475189, an in-content Follow button would be more ideal:

**(1) Posts:**
![screenshot-2016-07-28-15 31 50](https://cloud.githubusercontent.com/assets/4924246/17232656/7d526730-54de-11e6-9dbe-038387d9a3e5.png)

**(2) Site picks:**
![screenshot-2016-07-28-16 00 14](https://cloud.githubusercontent.com/assets/4924246/17232662/84632780-54de-11e6-9a02-0a5e0c24c6cd.png)

**(3) Site pick in full-post:**

![screenshot-2016-07-28-16 16 05](https://cloud.githubusercontent.com/assets/4924246/17232742/1590eca6-54df-11e6-8dae-f879b1456ea5.png)

For (3), we should just reuse the "Visit" style that's being used on the card on the main Discover stream. The blue bar is confusing as this style is being used for something else...

![screenshot 2016-07-28 16 21 33](https://cloud.githubusercontent.com/assets/4924246/17232817/a3eb3b00-54df-11e6-9a22-965b13e5ee94.png)

This should be used for limited excerpts only.

![screenshot 2016-07-28 16 21 41](https://cloud.githubusercontent.com/assets/4924246/17232818/a53656ac-54df-11e6-8209-1bbe47e168c9.png)

jhnstn commented:
@blowery @jancavan I reverted the post options follow button and added the inline follow buttons on the stream. 
I still have to do the site full-post

jhnstn commented:
@jancavan full-post is done

blowery commented:
reviewing

blowery commented:
fixed stats usage in https://github.com/Automattic/wp-calypso/pull/7103/commits/b67658df68e0f777aa852e9482ae6d76512a5b8b

jhnstn commented:
updated stats usage

jancavan commented:
If attribution is long, Follow button gets pushed down. Happens before the `660px` breakpoint:

![screenshot 2016-07-29 11 08 11](https://cloud.githubusercontent.com/assets/4924246/17258648/bffeac76-557c-11e6-93c2-c54f81e8c21e.png)

Note: The social icons get misaligned as well, but that's a separate issue. I'll fix in a separate PR.

Also in `<480px`:
![screenshot 2016-07-29 11 08 40](https://cloud.githubusercontent.com/assets/4924246/17258663/cfbc1e1e-557c-11e6-83e4-38c487ca7eda.png)

jancavan commented on client/reader/discover/_style.scss:
```diff
@@ -1,73 +1,89 @@
+		.noticon {
+			font-size: 20px;
+			height: 20px;
+			background-color: #c8d7e1;
```
Should be `lighten( $gray, 20% )`

jancavan commented on client/reader/discover/_style.scss:
```diff
@@ -1,73 +1,89 @@
+			font-size: 20px;
+			height: 20px;
+			background-color: #c8d7e1;
+			color: #fff;
```
$white

jancavan commented on client/reader/discover/_style.scss:
```diff
@@ -1,73 +1,89 @@
+		margin-top: 13px;
+		margin-bottom: 8px;
+
+		a {
```
There's a preference to use specific classnames instead of generic selectors, although I think in this case there are already specific classes for links under `.discover-attribution`?

jancavan commented on client/reader/discover/_style.scss:
```diff
@@ -1,73 +1,89 @@
-	}
+		&.is-missing-avatar {
+			&.is-post {
+				padding-left: 26px;
```
This `padding-left` adds indentation to some missing gravatars:

![screenshot 2016-07-29 11 32 30](https://cloud.githubusercontent.com/assets/4924246/17259406/27d805e2-5580-11e6-9b84-9769251142f3.png)

jancavan commented on client/reader/discover/_style.scss:
```diff
@@ -1,73 +1,89 @@
+		}
+
+		&.is-post a {
+			padding-left: 3px;
```
Probably better shorthand and also as a non-generic selector.

jhnstn commented:
@jancavan fixed viewport issue:
![screen](https://cloud.githubusercontent.com/assets/744755/17259510/b5559a38-5580-11e6-91d8-dbe63d3a4a5b.png)
![screen2](https://cloud.githubusercontent.com/assets/744755/17259522/c5a7161e-5580-11e6-8ddb-6df3141158d2.png)

jancavan commented:
In https://github.com/Automattic/wp-calypso/pull/7103/commits/f89ef52bd410afd8e0a021e253fd6ae32807b3b1:

- I don't think we need to hide Follow button text in narrow viewports so unhid it
- Fixed Follow button indenting in narrow viewports
![screenshot 2016-07-29 14 19 06](https://cloud.githubusercontent.com/assets/4924246/17264297/dce2f176-559a-11e6-8cb9-64be5763970b.png)

blowery commented:
looks good, 🚢

fraying commented:
I love this. (It's something I recommended way back in my trial.) Well done!
